src/org/daisy/dotify/formatter/impl/page/PageSequenceBuilder2.java

```diff
@@ -363,6 +364,7 @@ private PageImpl nextPageInner(
                     master.getFlowWidth() -
                     master.getTemplate(current.getPageNumber()).getTotalMarginRegionWidth()
                 )
+                .topOfPage(true)
```

Comment:
Have changed so we could ensure that it's at the beginning of a page. Will look into if there is more to do to handle blocks.

Reply:
Right. So the first condition is done correctly now, but we don't know anything about the second condition ("we are at the beginning of the block") at the moment.

We discussed adding a test with a block that could potentially be split across pages.

There is another test case I just thought of: what if display-when="$starts-at-top-of-page" (so without the negation!) and the block has more than one row. The way it is implemented now I expect that when this is block is the first of the page, the first row will be displayed but the second not.

A possible solution for this problem is to disallow another else but `(! $starts-at-top-of-page)`.

Reply:
I've changed the code and added a test for that.

Reply:
Sorry to disappoint you, see [other comment](https://github.com/mtmse/dotify.formatter.impl/pull/36#issuecomment-739875491).

Reply:
It's still bugging me a bit that the `testTwoBlocksWithDisplayWhenAsTrue()` test passes (because of the 31534105c933de51713ac6bf8b2f2e912eb69a92 change) but that it proves nothing. Because it's potentially confusing it does more hard than good IMO. More useful would be if we expect it to fail. Then it would actually prove something, namely that the implementation is limited (has undefined behavior when it receives unexpected input).